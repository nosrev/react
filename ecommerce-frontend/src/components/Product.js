import React from 'react';
import image from '../img/produto.png';
import { Link } from 'react-router-dom';

const Product = ({type}) => {
    if (type === "home" || type === "popular") {
        return (
            <Link to='/product'>
                <div className="product border border-slate-300">
                    <span className="product-badge">oferta</span>
                    <img src={image} alt="Camiseta algodão" />
                    <p className="product-name">Camiseta algodão</p>
                    <p className="product-price">
                        <span className="old-price">R$200.00</span>
                        <span className="special-price">R$100.00</span>
                    </p>
                </div>
            </Link>
        );
    }

    return (
        <Link to='/product'>
            <div className="product border border-slate-300">
                <img src={image} alt="Camiseta algodão" />
                <p className="product-name">Camiseta algodão</p>
                <p className="product-price">R$175.00</p>
                <p className="short-desc">
                    Camiseta Comfort Básica em Algodão com Manga Curta<br/>
                    Camiseta comfort, em algodão peruano, com mangas curtas. Confeccionada em algodão peruano, a peça é leve e respirável.
                </p>
            </div>
        </Link>
    );
};

export default Product;
