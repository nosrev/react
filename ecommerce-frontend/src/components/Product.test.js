import renderer from 'react-test-renderer';
import Product from './Product';
import { MemoryRouter as Router } from 'react-router-dom';

it('render Product', () => {
  const component = renderer.create(
    <Router>
        <Product type="home" />,
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
