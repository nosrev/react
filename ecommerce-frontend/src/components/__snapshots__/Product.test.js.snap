// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Product 1`] = `
Array [
  <a
    href="/product"
    onClick={[Function]}
  >
    <div
      className="product border border-slate-300"
    >
      <span
        className="product-badge"
      >
        oferta
      </span>
      <img
        alt="Camiseta algodão"
        src="produto.png"
      />
      <p
        className="product-name"
      >
        Camiseta algodão
      </p>
      <p
        className="product-price"
      >
        <span
          className="old-price"
        >
          R$200.00
        </span>
        <span
          className="special-price"
        >
          R$100.00
        </span>
      </p>
    </div>
  </a>,
  ",",
]
`;
